### PR TITLE
Add gstream

### PR DESCRIPTION
**Pull request by biozit**

### CODE REVIEW

**Comment by djw8605** [comment]
I would like to see a cleaner separation of the decoding vs. the "enriching".  So in the decoding.py file, only have code that decodes the binary packet to the tuple (which I think existed before).  Then the json parsing should be in the DetailedCollector file.  That's how we have separated the logic so far.

You should make a separate function in the DetailedCollector that enriches and extracts the json components.  Much like the tcp parsing.

File: Collectors/DetailedCollector.py [hunk 1]
```diff
@@ -374,15 +382,51 @@ def process(self, data, addr, port):
             pass
 
         elif header.code == b'g':
-            # The rest of the message is the gstream event
             self.logger.debug("Received gstream message")
+
+            userfromserver = self._users[sid].items()[0];
```
Comment (djw8605) [inline Collectors/DetailedCollector.py:387]: I think you will need to check if the `sid` exists in the `_users` dict before trying to access it.  Also, this is not how you get the user information.  You get the user information like the following:

https://github.com/opensciencegrid/xrootd-monitoring-collector/blob/ab8a2af1b48bf3372bcaebcba8d66fd3b2822b80/Collectors/DetailedCollector.py#L65-L70

File: Collectors/DetailedCollector.py [hunk 2]
```diff
@@ -228,6 +228,14 @@ def addRecord(self, sid, userID, fileClose, timestamp, addr, openTime):
             self.metrics_q.put({'type': 'message sent', 'count': 1, 'message_type': 'wlcg'})
 
         return rec
+        
+    # return the VO based on the on the path    
+    def returnVO(fname):
```
Comment (djw8605) [inline Collectors/DetailedCollector.py:233]: What is this function suppose to do?  It seems to only work with icecube?

**Review by djw8605 — CHANGES_REQUESTED** [review]
Some comments.

**Comment by biozit** [comment]
sorry for this, this should be only for my fork. I am still working on it.

**Comment by djw8605** [comment]
No worries.  Yup, making it a draft is the correct move.  Let me know when you want me to review it.

**Pull request closed without merging**